File: Tests/AcaiParserGoldenTests/Fixtures/Shop.java
package com.example.shop;

import java.util.ArrayList;
import java.util.List;

interface Priced {
    double price();
}

enum Category {
    FOOD,
    TOOL;

    public String label() {
        return name().toLowerCase();
    }
}

record Money(double amount, String currency) {
}

abstract class Item implements Priced {

    protected final String name;

    Item(String name) {
        this.name = name;
    }

    public abstract double price();

    public String describe() {
        return this.name;
    }
}

public class Product extends Item {

    private final Logger audit = new Logger();
    private double base;
    public double discount;
    private final List<String> tags = new ArrayList<>();

    public static class Builder {
        private String name = "";

        public Product build() {
            return new Product(this.name, 0.0);
        }
    }

    public Product(String name, double base) {
        super(name);
        this.base = base;
    }

    @Override
    public double price() {
        this.audit.log("pricing");
        return this.base - this.discount;
    }

    @Override
    public String describe() {
        Logger local = new Logger();
        local.log(this.name);
        return this.name;
    }

    public Category classify(int value) {
        if (value > 10) {
            return Category.TOOL;
        } else if (value > 5) {
            return Category.FOOD;
        }
        for (int index = 0; index < value; index++) {
            this.discount += index;
        }
        return Category.FOOD;
    }

    public <T> T identity(T value) {
        return value;
    }
}

class Logger {
    void log(String message) {
    }
}

File: Tests/AcaiParserGoldenTests/Fixtures/shop.cpp
#include <string>
#include <vector>

namespace shop {

enum class Category {
    Food,
    Tool
};

struct Money {
    double amount;
    std::string currency;
};

class Logger {
public:
    void log(const std::string &message);
};

class Priced {
public:
    virtual double price() const = 0;
    virtual ~Priced() = default;
};

class Item : public Priced {
protected:
    std::string name;

public:
    explicit Item(const std::string &name);
    virtual std::string describe() const;
};

class Product : public Item {
private:
    Logger audit;
    double base;
    std::vector<std::string> tags;

public:
    double discount;
    static const std::string defaultCurrency;

    class Builder {
    public:
        Product build() const;
    };

    Product(const std::string &name, double base);

    double price() const override {
        return base - discount;
    }

    std::string describe() const override {
        return name;
    }

    Category classify(int value) {
        if (value > 10) {
            return Category::Tool;
        } else if (value > 5) {
            return Category::Food;
        }
        for (int index = 0; index < value; ++index) {
            discount += index;
        }
        return Category::Food;
    }

    template <typename T>
    T identity(T value) const {
        return value;
    }
};

double topLevelDiscount(const Product &product) {
    return product.price();
}

}  // namespace shop
